packages/mds-utils/utils.ts

```diff
@@ -535,9 +535,9 @@ function clone<T>(obj: T): T {
 }
 
 // T is the non-null and non-undefined type
-function filterEmptyHelper<T>(warnOnEmpty?: boolean) {
+function isDefined<T>(warnOnEmpty?: boolean) {
```

Comment:
You could make a couple of functions:
```ts
isDefined = <T>(value: T | undefined | null, warnOnEmpty = false): value is T => {...}
```
And then:
```ts
filterDefined = (warnOnEmpty = false) => <T>(value: T | undefined | null, index: number, array: (T | undefined | null)[]): value is T => isDefined(value, warnOnEmpty)
```
This would allow for both:
```ts
if (isDefined(value)) {
  // instead of isDefined()(value)
}
```
And
```ts
values.filter(filterDefined())
```